report/renderer/util.js

```diff
@@ -72,6 +72,21 @@ class Util {
     return lhr.mainDocumentUrl || lhr.finalUrl;
   }
 
+  /**
+   * @param {string} url
+   * @return {?string}
+   */
+  static getOrigin(url) {
+    try {
+      const urlInfo = new URL(url);
+      // check for both host and origin since some URLs schemes like data and file set origin to the
+      // string "null" instead of the object
+      return (urlInfo.host && urlInfo.origin) || null;
```

Comment:
If this isn't literally "give me the url origin", then this function needs a different name.

Also I think `??` is better here than `&&`?

Reply:
This function is a copy of `getOrigin` from `core/util.cjs` to make it available for reporting side. It looks like we have multiple copies of this function, `utils.cjs`, `url-utils.js` and now in `reports/render/util.js`. 

Re: `??`, null coalescing operator would return 1st operand when it's non-null, i.e., in this case it would end up returning host instead of origin, which is not what we want here I think?

Reply:
hmmm didn't catch this was an existing function, so nevermind


i guess I find idiosyncratic JS usage of `&&` as shorthand incmhrensible(sp ;) then cuz I'm still not getting it here (altho it's rather early for me right now..)

Reply:
☕ 

We probably need to consolidate this URL functions at some point - so we could share between `core/audit*` and `report/*` side of things, vs. duplicating fn and its unit tests.

Reply:
#14378 was that

Reply:
This function is not needed with the latest refactor, so removed it.